plugin/pkg/project/validator/admission.go

```diff
@@ -56,10 +59,65 @@ func (v *handler) Validate(_ context.Context, a admission.Attributes, _ admissio
 		return apierrors.NewBadRequest("could not convert object to Project")
 	}
 
-	// TODO: Remove this admission plugin in favor of static validation in a future release, see https://github.com/gardener/gardener/pull/4228.
+	// TODO: Remove this check in favor of static validation in a future release, see https://github.com/gardener/gardener/pull/4228.
 	if project.Spec.Namespace != nil && *project.Spec.Namespace != v1beta1constants.GardenNamespace && !strings.HasPrefix(*project.Spec.Namespace, gardenerutils.ProjectNamespacePrefix) {
 		return admission.NewForbidden(a, fmt.Errorf(".spec.namespace must start with %s", gardenerutils.ProjectNamespacePrefix))
 	}
 
+	if utils.SkipVerification(a.GetOperation(), project.ObjectMeta) {
+		return nil
+	}
+
+	if a.GetOperation() == admission.Create {
+		ensureProjectOwner(project, a.GetUserInfo().GetName())
+	}
+
+	ensureOwnerIsMember(project)
+
 	return nil
 }
+
+func ensureProjectOwner(project *gardencore.Project, userName string) {
+	// Set createdBy field in Project
+	project.Spec.CreatedBy = &rbacv1.Subject{
+		APIGroup: "rbac.authorization.k8s.io",
+		Kind:     rbacv1.UserKind,
+		Name:     userName,
+	}
+
+	if project.Spec.Owner == nil {
+		owner := project.Spec.CreatedBy
+
+	outer:
+		for _, member := range project.Spec.Members {
+			for _, role := range member.Roles {
+				if role == gardencore.ProjectMemberOwner {
+					owner = member.Subject.DeepCopy()
+					break outer
+				}
+			}
+		}
+
+		project.Spec.Owner = owner
```

Comment:
Nit: By extracting the code into a function we could get rid of the `break` with label. The function does not have to be anonymous, but could also have parameters.
WDYT?

```suggestion
		project.Spec.Owner = func() *rbacv1.Subject {
			for _, member := range project.Spec.Members {
				for _, role := range member.Roles {
					if role == gardencore.ProjectMemberOwner {
						return member.Subject.DeepCopy()
					}
				}
			}
			return project.Spec.CreatedBy
		}()
```